fix: update sonarqube depend version to fix vul.

kycheng opened the pull request:
Part of 
<!--
  Only for standalone PRs without Jira issue in the PR title:
    * Replace this comment with Epic ID to create a new Task in Jira
    * Replace this comment with Issue ID to create a new Sub-Task in Jira
    * Ignore or delete this note to create a new Task in Jira without a parent
-->

kycheng committed:
fix: update various dependencies and improve database index definitions

lentil1016 commented on build.gradle:
```diff
@@ -218,26 +237,26 @@ subprojects {
+      dependency 'org.sonarsource.javascript:sonar-javascript-plugin:10.16.0.27621'
       dependency 'org.sonarsource.php:sonar-php-plugin:3.27.1.9352'
-      dependency 'org.sonarsource.python:sonar-python-plugin:3.24.1.11916'
+      dependency 'org.sonarsource.python:sonar-python-plugin:4.24.0.18631'
```
这个python插件的修改可能导致了一些问题，周日有一些流水线回归失败了
回归版本：3.20.6-hotfix.1127.1.g333ed60e-fix-vul-320
![image](https://github.com/user-attachments/assets/ee2f3884-f893-4043-a1f8-dc3c84cbf0c4)
<img width="1340" alt="image" src="https://github.com/user-attachments/assets/9a9f0b28-3d1b-4f1d-bab4-3ca208073fd5" />

kycheng committed:
feat: update commons-io and sonar-javascript-plugin dependencies

- Added resolution strategy to force commons-io version 2.14.0 and exclude previous versions.
- Updated sonar-javascript-plugin to version 10.18.0.28572 for improved functionality.
